# Anise & Basil

BASIL is a method to detect breakpoints for structural variants (including insertion breakpoints) from aligned paired HTS reads in BAM format.
ANISE is a method for the assembly of large insertions from paired reads in BAM format and a list candidate insert breakpoints as generated by BASIL.

## Quickstart

### Obtaining and Compiling

For obtaining the software, please use git for downloading the latest stable version from the *master* branch.

```
# git clone git@github.com:seqan/anise_basil.git
```

Then, use `git submodule` for fetching the SeqAn library dependency.

```
# cd anise_basil
# git submodule init
# git submodule update
```

Finally, compile the program.
For this, you will need [CMake] and a C++ compiler with C++11 support.
If you have a fairly recent Linux distribution or Mac Os X with a fairly recent Xcode version, you should be all set.

```
# cd build
# cmake ..
# make -j 4
```

You can control the number of cores to use for compiling with the `-j` parameter to `make`, thus the code above gives an example for the compilation with 4 cores.
You can now have a look at the command line help for both programs:

```
# ./bin/basil -h
# ./bin/anise -h
```

### A First Example

First, use BASIL for the prediction of insert locations, given the reference file `ref.fa` and the alignment file `in.bam` that can be found in the `examples` directory.

```
# ./bin/basil --in-reference examples/ref.fa \
    --in-alignment examples/in.bam \
    --out-vcf basil.vcf
```

Then, use ANISE for assembling the inserted sequences.

```
# ./bin/anise --in-reference examples/ref.fa \
    --in-alignment examples/in.bam \
    --in-vcf basil.vcf \
    --out-sequences anise.fa
```

## References

* Holtgrewe, M., Kuchenbecker, L., & Reinert, K. (2015).
  [Methods for the Detection and Assembly of Novel Sequence in High-Throughput Sequencing Data].
  *Bioinformatics*, btv051.

## Contact

For questions, comments, or suggestions, please file a [GitHub issue] or an email to [Manuel Holtgrewe]

[CMake]: http://www.cmake.org
[Methods for the Detection and Assembly of Novel Sequence in High-Throughput Sequencing Data]: http://bioinformatics.oxfordjournals.org/content/early/2015/02/01/bioinformatics.btv051.short
[GitHub issue]: https://github.com/seqan/anise_basil/issues
[Manuel Holtgrewe]: mailto:manuel.holtgrewe@fu-berlin.de
